src/events.h

```diff
@@ -393,6 +395,14 @@ struct FWorldEvent
 	bool DamageIsRadius; // radius damage yes/no
 	int NewDamage = 0; // sector/line damaged. allows modifying damage
 	FState* CrushedState = nullptr; // custom crush state set in thingground
+	DVector3 AttackPos; //hitscan point of origin
+	DAngle AttackAngle;
+	DAngle AttackPitch;
+	double AttackDistance;
+	double AttackOffsetForward;
+	double AttackOffsetSide;
+	double AttackZ;
+	PClassActor* AttackPuffType;
```

Comment:
`double`s and pointers need default initialization values, otherwise they'll be corrupted